river/XwaylandView.zig

```diff
@@ -193,12 +214,21 @@ fn handleCommit(listener: ?*c.wl_listener, data: ?*c_void) callconv(.C) void {
 
     // See comment in XwaylandView.configure()
     if (view.pending_serial != null) {
+        // Update the stored dimensions of the surface
+        view.surface_box = new_box;
+
         // If the view is part of the layout, notify the transaction code. If
         // the view is floating or fullscreen apply the pending state immediately.
         view.pending_serial = null;
-        if (!view.pending.float and !view.pending.fullscreen)
+        if (view.shouldTrackConfigure())
             view.output.root.notifyConfigured()
-        else
+        else {
+            const view_tags_changed = view.pending.tags != view.current.tags;
             view.current = view.pending;
+            view.commitOpacityTransition();
+            if (view_tags_changed) view.output.sendViewTags();
+        }
```

Comment:
This is exactly the same as what is in `XdgToplevel.zig`, we should probably extract it to a new function in `View.zig`